dali/imgcodec/parsers/jpeg2000.cc

```diff
@@ -13,18 +13,57 @@
 // limitations under the License.
 
 #include "dali/imgcodec/parsers/jpeg2000.h"
+#include "dali/core/byte_io.h"
 
 namespace dali {
 namespace imgcodec {
 
+namespace {
+
+using block_type_t = std::array<uint8_t, 4>;
+const block_type_t jp2_sig_type = {'j', 'P', ' ', ' '};
+const block_type_t jp2_format_type = {'f', 't', 'y', 'p'};
+const block_type_t jp2_header_type = {'j', 'p', '2', 'h'};
+const block_type_t jp2_im_header_type = {'i', 'h', 'd', 'r'};
+
+void advance_one_block(InputStream &stream) {
+  const auto block_size = ReadValueBE<uint32_t>(stream);
+  // skipping the rest of the block
+  stream.Skip(block_size - sizeof(block_size));
+}
+
+}  // namespace
+
 ImageInfo Jpeg2000Parser::GetInfo(ImageSource *encoded) const {
+  auto stream = encoded->Open();
+
+  advance_one_block(*stream);  // jp2_sig_type
+  advance_one_block(*stream);  // jp2_format_type
+
+  // jp2 header starts, we skip:
+  // block size
+  // block type (jp_header_type)
+  // image block size
+  // image block type  (jp_im_header_type)
+  stream->Skip<std::array<uint32_t, 4>>();
```

Comment:
```suggestion
  stream->Skip<uint32_t>(4);
```
would be more natural